api/src/main/scala/za/co/absa/loginsvc/rest/AuthManagerConfig.scala

```diff
@@ -33,34 +33,34 @@ import scala.collection.immutable.SortedMap
 @Configuration
 class AuthManagerConfig @Autowired()(authConfigProvider: AuthConfigProvider){
 
-  private val usersConfig: UsersConfig = authConfigProvider.getUsersConfig
-  private val adLDAPConfig: ActiveDirectoryLDAPConfig = authConfigProvider.getLdapConfig
+  private val usersConfig: Option[UsersConfig] = authConfigProvider.getUsersConfig
+  private val adLDAPConfig: Option[ActiveDirectoryLDAPConfig] = authConfigProvider.getLdapConfig
 
   private val logger = LoggerFactory.getLogger(classOf[AuthManagerConfig])
 
   @Bean
   def authManager(http: HttpSecurity): AuthenticationManager = {
 
     val authenticationManagerBuilder = http.getSharedObject(classOf[AuthenticationManagerBuilder]).parentAuthenticationManager(null)
-    val configs: Array[DynamicAuthOrder] = Array(usersConfig, adLDAPConfig)
+    val configs: Array[DynamicAuthOrder] = Array(usersConfig, adLDAPConfig).flatten
     val orderedProviders = createProviders(configs)
 
     if(orderedProviders.isEmpty)
       throw ConfigValidationException("No authentication method enabled in config")
 
-    orderedProviders.foreach(
+    orderedProviders.zipWithIndex.foreach(
       auth => {
-        logger.info(s"Authentication method ${auth._2.getClass.getSimpleName} has been initialized at order ${auth._1}")
-        authenticationManagerBuilder.authenticationProvider(auth._2)
+        logger.info(s"Authentication method ${auth._1.getClass.getSimpleName} has been initialized at order ${auth._2 + 1}")
+        authenticationManagerBuilder.authenticationProvider(auth._1)
       })
```

Comment:
While technically possible, using `tuple._1` and `tuple._2` is hard to read. The accepted pattern here is to pattern-match to name the fields:

```scala
    orderedProviders.zipWithIndex.foreach { case (authProvider, index) =>
      logger.info(s"Authentication method ${authProvider.getClass.getSimpleName} has been initialized at order ${index + 1}")
      authenticationManagerBuilder.authenticationProvider(authProvider)
    }
```

Reply:
Thank you. Will keep this in mind in future.